libbeat/publisher/pipeline/pipeline.go

```diff
@@ -386,3 +367,36 @@ func (e *waitCloser) dec(n int) {
 func (e *waitCloser) wait() {
 	e.events.Wait()
 }
+
+func makePipelineProcessors(
+	annotations Annotations,
+	processors *processors.Processors,
+	disabled bool,
+) pipelineProcessors {
+	p := pipelineProcessors{}
+
+	hasProcessors := processors != nil && len(processors.List) > 0
+	if hasProcessors {
+		tmp := &program{title: "global"}
+		for _, p := range processors.List {
+			tmp.add(p)
+		}
+		p.processors = tmp
+	}
+
+	fields := common.MapStr{}
```

Comment:
I suggest to move this to line 391 as it is only used there.

Reply:
Yeah. Will simplify fields initialization.